dsp-repository/v1/dsp-repository.shacl.ttl

```diff
@@ -703,3 +696,14 @@ dsp-repo:DataManagementPlanShape
     sh:closed true ;
     sh:ignoredProperties (rdf:type) .
 
+### --- Shape of prov:Attribution ---
+dsp-repo:AttributionShape
+    a sh:NodeShape ;
+    sh:targetClass prov:Attribution ;
+    sh:property [
+        sh:path dsp-repo:hasRole ;
+        sh:name "Role" ;
+        sh:description "Role of the person within the dataset"@en ;
+        sh:datatype xsd:string ;
+        sh:minCount 1 ;
+    ] .
```

Comment:
`AttributionShape` should have two Properties: `hasRole` and `agent`.  
Do we need to add the `agent` here as well?

Furthermore, should this maybe be a closed shape?

And finally: If we define the shape here, would we need to define the class itself in the beginning of the file? Or is that not necessary, because it's a `prov:` class?

Reply:
I can answer all your questions with a "No" ;-)
We do not need to add `agent` to `AttributionShape` because `agent` is in `prov:` already defined.

The shape has to stay open because we did not define the property by ourselves. It is a `prov:` property.

Also not necessary, it is a `prov:` class, which is already defined.

Conclusion: We just love `prov:` :-) But maybeee we do need a SHACL-shape for `prov:` and for `schema:`

Reply:
In that case: Wonderful!

> But maybeee we do need a SHACL-shape for prov: and for schema:
Could you add something along these lines to DSP-954?